### PR TITLE
Retry logic added in KeyVault task

### PR DESCRIPTION
**Pull request by tauhid621**
**Task name**: AzureKeyVaultV1

**Description**: Retry logic added in case of failure.

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) #13998 

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts [hunk 1]
```diff
@@ -28,21 +29,45 @@ export class KeyVaultClient extends azureServiceClient.ServiceClient {
...
-            var response = await this.beginRequest(request);
-            if (response.statusCode == 401) {
-                var vaultResourceId = this.getValidVaultResourceId(response);
-                if(!!vaultResourceId) {
-                    console.log(tl.loc("RetryingWithVaultResourceIdFromResponse", vaultResourceId));
-                    
-                    this.getCredentials().activeDirectoryResourceId = vaultResourceId; // update vault resource Id
-                    this.getCredentials().getToken(true); // Refresh authorization token in cache
-                    var response = await this.beginRequest(request);
+        const maxRetryCount: number = 5;
```
Comment (ds-ms) [inline Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts:32]: Doesn't the webclient take all these as parameters? https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/azure-arm-rest-v2/webClient.ts#L39

Comment (tauhid621) [reply Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts:32]: We are not calling the web client directly. we are using '[beginRequest](https://github.com/microsoft/azure-pipelines-tasks/blob/4c3de8b30f2bf5a647d124254eae95a2e1f0698d/common-npm-packages/azure-arm-rest-v2/AzureServiceClientBase.ts#L105)' of azure service client which calls webclient.

File: Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts [hunk 2]
```diff
@@ -28,21 +29,45 @@ export class KeyVaultClient extends azureServiceClient.ServiceClient {
...
+        const retryIntervalInSeconds: number = 2;
+        const retriableErrorCodes = ["ETIMEDOUT", "ECONNRESET", "ENOTFOUND", "ESOCKETTIMEDOUT", "ECONNREFUSED", "EHOSTUNREACH", "EPIPE", "EA_AGAIN", "EAI_AGAIN"];
+        const retriableStatusCodes = [408, 409, 500, 502, 503, 504];
+        let timeToWait: number = retryIntervalInSeconds;
+        let retryCount: number = 0;
+        
+        while(true) {
```
Comment (ds-ms) [inline Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts:39]: Have a max number of retries maybe?

Comment (ds-ms) [reply Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts:39]: `while(++retryCount < maxRetryCount)` instead maybe

**Review by ds-ms — APPROVED** [review]
LGTM! Do test out an e2e before merging

File: Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts (hunk 2 above)
Comment (tauhid621) [reply Tasks/AzureKeyVaultV1/operations/azure-arm-keyvault.ts:39]: retry logic is inside the loop

**Pull request merged**